### PR TITLE
[jaeger-v2] Add HotROD integration test for jaeger-v2

### LINKED ISSUE #6137: Add HotROD integration test for jaeger-v2
An issue was reported in #6134 which highlights the fact that we're not CI-testing examples/hotrod/docker-compose-v2.yml.

We need to extend `.github/workflows/ci-docker-hotrod.yml` to run a matrix with both v1 and v2 versins, similar to `.github/workflows/ci-e2e-spm.yml` or `.github/workflows/ci-e2e-cassandra.yml`.

### PR DESCRIPTION
**Pull request by Saumya40-codes**
## Which problem is this PR solving?
- Resolves #6137 

## Description of the changes
- Currently in `.github/workflows/ci-docker-hotrod.yml`, we are only testing internally via `.scripts/build-hotrod-image.sh` for `examples/hotrod/docker-compose.yml` while `examples/hotrod/docker-compose-v2.yml` isnt getting covered

- For this, in `.github/workflows/ci-docker-hotrod.yml` we are passing `-b` flag to specify the jaeger binary (`all-in-one` or `jaeger`)

## How was this change tested?
- For the flags and normal bash scirpt code i tested it manually in seperate bash file so that atleast ensures the basic code test

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: .github/workflows/ci-docker-hotrod.yml [hunk 1]
```diff
@@ -18,6 +18,10 @@ permissions:
 jobs:
   hotrod:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        jaeger-version: [v1, v2]
```
Comment (Saumya40-codes) [inline .github/workflows/ci-docker-hotrod.yml:24]: should we specify binary here instead?
as `examples/hotrod/docker-compose-v2.yml` depends on `jaeger` (v2) image and `examples/hotrod/docker-compose.yml` on `all-in-one` (v1)

**Comment by Saumya40-codes** [comment]
v2 test passes now, though i'd one doubt regarding execution which i'll just left in comments

File: scripts/build-hotrod-image.sh [hunk 2]
```diff
@@ -73,9 +85,16 @@ done
...
-make build-all-in-one
-bash scripts/build-upload-a-docker-image.sh -l -b -c all-in-one -d cmd/all-in-one -p "${current_platform}" -t release
+# Build all-in-one (for v1) or jaeger (for v2) image locally (-l) for integration test
+if [[ "$BINARY" == "all-in-one" ]]; then
+    # Build all-in-one image for v1
+    make build-all-in-one
+    bash scripts/build-upload-a-docker-image.sh -l -b -c all-in-one -d cmd/all-in-one -p "${current_platform}" -t release
+else
+    # Build jaeger image for v2
+    make build-jaeger
```
Comment (Saumya40-codes) [inline scripts/build-hotrod-image.sh:95]: we should be keeping `make build-all-in-one` for both the cases, right?

File: .github/workflows/ci-docker-hotrod.yml [hunk 3]
```diff
@@ -18,6 +18,15 @@ permissions:
...
   hotrod:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        mode:
+        - name: v1
+          binary: all-in-one
+        - name: v2
+          binary: jaeger
```
Comment (yurishkuro) [inline .github/workflows/ci-docker-hotrod.yml:28]: please keep only the version in the matrix, and assign binary name inside the script

File: scripts/build-hotrod-image.sh [hunk 4]
```diff
@@ -6,21 +6,23 @@
 set -euf -o pipefail
 
 print_help() {
-  echo "Usage: $0 [-h] [-l] [-o] [-p platforms]"
+  echo "Usage: $0 [-h] [-l] [-o] [-p platforms] [-b binary]"
```
Comment (yurishkuro) [inline scripts/build-hotrod-image.sh:9]: accept `-v` instead, similar to most other scripts

File: scripts/build-hotrod-image.sh (hunk 2 above)
Comment (yurishkuro) [reply scripts/build-hotrod-image.sh:95]: no, what you have is correct, v2 is a different binary

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
Thanks!

**Pull request merged**